src/scripts.js

```diff
@@ -76,54 +65,111 @@ function renderApplication() {
     formDestinations.innerHTML += `<option value="${destinationName}">${destinationName}</option>`;
   });
-  // show destinations in the main content?
 }
 
-function renderTripsForUser() {
+const renderTripsForUser = () => {
   fetchData('trips')
     .then(data => {
       const traveler = new Traveler(randomUser, data.trips);
-
+      pastTrips.innerHTML = '<h3>Past Trips</h3>';
+      upcomingTrips.innerHTML = '<h3>Upcoming Trips</h3>';
+      pendingTrips.innerHTML = '<h3>Pending Trips</h3>';
       traveler.listOfTrips.forEach(userTrip => {
         const trip = new Trip(userTrip);
         const tripDestination = allDestinations.getDestinationById(trip.destinationID);
 
         if (trip.isPastTrip() && !trip.isPendingTrip()) {
-          pastTrips.innerHTML += `<button class="trip">${tripDestination.destination} on ${trip.date}</button>`;
+          pastTrips.innerHTML += `<button class="trip-btn">${tripDestination.destination} on ${trip.date}</button>`;
         } else if (trip.isUpcomingTrip() && !trip.isPendingTrip()) {
-          upcomingTrips.innerHTML += `<button class="trip">${tripDestination.destination} on ${trip.date}</button>`;
+          upcomingTrips.innerHTML += `<button class="trip-btn">${tripDestination.destination} on ${trip.date}</button>`;
         } else if (trip.isPendingTrip()) {
-          pendingTrips.innerHTML += `<button class="trip">${tripDestination.destination} on ${trip.date}</button>`;
+          pendingTrips.innerHTML += `<button class="trip-btn">${tripDestination.destination} on ${trip.date}</button>`;
         }
       });
     }).catch(err => console.log('There was a problem retrieving your data', err))
 }
 
-function renderYearlySpending() {
+const renderYearlySpending = () => {
   fetchData('trips')
     .then(data => {
       const traveler = new Traveler(randomUser, data.trips)
-      console.log(traveler)
       const yearlySpending = traveler.getYearlySpendingOnTrips(allDestinations);
 
       userSpending.innerText = `Total Spending this past year: $${yearlySpending}`;
     }).catch(err => console.log('There was a problem retrieving your data', err))
 }
 
+const changeDateFormat = (date) => {
+  return date.split('-').join('/');
+}
+
+const handleFormSubmit = (event) => {
+  event.preventDefault();
+  fetchData('trips')
+    .then(data => {
+      let tripsLength = data.trips.length;
+      let formData = {
+        id: tripsLength + 1,
+        userID: randomUser.id,
+        destinationID: allDestinations.getDestinationIdByName(formDestination.value),
+        travelers: parseInt(formNumOfTravelers.value),
+        date: changeDateFormat(formTripDate.value),
+        duration: parseInt(formDuration.value),
+        status: 'pending',
+        suggestedActivities: [],
+      }
+      return formData;
+    }).then(formData => {
+      postTripRequest(formData);
+      renderApplication();
+      event.target.reset();
+    })
+}
+
+const handleCostEstimate = () => {
```

Comment:
Confused why the cost estimate has a fetch in it? Could this just be a method on one of the classes? And instead use the fetch call once to get all the data and make new instances of each of the trips/destinations there.

Reply:
When handleCostEstimate is called, it needs to re-fetch the information from the form based on the user's input. I'm glad you pointed this out because I realize I don't need to formData in this case to calculate the cost estimate.